app/assets/javascripts/discourse/app/components/topic-list/topic-list-item.gjs

```diff
@@ -24,15 +25,34 @@ import discourseTags from "discourse/helpers/discourse-tags";
 import formatDate from "discourse/helpers/format-date";
 import number from "discourse/helpers/number";
 import topicFeaturedLink from "discourse/helpers/topic-featured-link";
+import DAG from "discourse/lib/dag";
 import { wantsNewWindow } from "discourse/lib/intercept-click";
+import { applyValueTransformer } from "discourse/lib/transformer";
 import DiscourseURL from "discourse/lib/url";
 import icon from "discourse-common/helpers/d-icon";
 import i18n from "discourse-common/helpers/i18n";
 
+export function createColumns() {
+  const columns = new DAG();
+  columns.add("topic-list-before-columns");
+  columns.add("bulk-select");
+  columns.add("topic");
+  columns.add("topic-list-after-main-link");
+  columns.add("posters");
+  columns.add("replies");
+  columns.add("likes");
+  columns.add("op-likes");
+  columns.add("views");
+  columns.add("activity");
+  columns.add("topic-list-after-columns");
+  return columns;
+}
+
```

Comment:
Seems a little strange that we have this re-implemented in both topic-list-header and topic-list-item. I imagine people will often want to customize both of these things in the same way... so wouldn't it make sense for them to be in the same DAG?